modules/Xsens_module/src/XsensRetargeting.cpp

```diff
@@ -56,10 +66,23 @@ bool XsensRetargeting::configure(yarp::os::ResourceFinder& rf)
     }
 
     yarp::os::Value* axesListYarp;
-    if (!rf.check("joints_list", axesListYarp))
+    if (m_yogaRetargeting)
+    { // read yoga retargeting
+        if (!rf.check("joints_list_yoga_controller", axesListYarp))
+        {
+            yError() << "[XsensRetargeting::configure] Unable to find joints_list_yoga_controller "
+                        "into config file.";
+            return false;
+        }
+    } else
     {
-        yError() << "[XsensRetargeting::configure] Unable to find joints_list into config file.";
-        return false;
+        // read from walking retargeting
+        if (!rf.check("joints_list_walking_controller", axesListYarp))
+        {
+            yError() << "[XsensRetargeting::configure] Unable to find "
+                        "joints_list_walking_controller into config file.";
+            return false;
+        }
```

Comment:
You can run the Module with two different configuration files. This will avoid having a different portion of the code for the `yogaRetargeting`

Reply:
done, thank you for your good suggestion.